src/interpreter_funcs/voxel_interpolated_union.rs

```diff
@@ -217,30 +214,40 @@ impl Func for FuncInterpolatedUnion {
         }
 
         let mut voxel_cloud1 = ScalarField::from_mesh(mesh1, &voxel_dimensions, 0.0, 1);
-        let voxel_cloud2 = ScalarField::from_mesh(mesh2, &voxel_dimensions, 0.0, 1);
+        let mut voxel_cloud2 = ScalarField::from_mesh(mesh2, &voxel_dimensions, 0.0, 1);
 
-        let boolean_union_range = 0.0..=0.0;
-
-        let meshing_range = if fill {
+        let volume_value_range = if fill {
             (Bound::Unbounded, Bound::Included(0.0))
         } else {
             (Bound::Included(0.0), Bound::Included(0.0))
         };
 
-        voxel_cloud1.interpolated_union_of_distance_fields(
-            &boolean_union_range,
-            &voxel_cloud2,
-            &boolean_union_range,
-            factor,
-        );
+        let bounding_box_voxel_cloud1 = voxel_cloud1.bounding_box_cartesian_space();
+        let bounding_box_voxel_cloud2 = voxel_cloud2.bounding_box_cartesian_space();
+
+        if let Some(bounding_box) = BoundingBox::union(
+            [bounding_box_voxel_cloud1, bounding_box_voxel_cloud2]
+                .iter()
+                .copied(),
+        ) {
+            voxel_cloud1.resize_to_bounding_box_cartesian_space(&bounding_box);
+            voxel_cloud1.compute_distance_field(&volume_value_range);
+            voxel_cloud2.compute_distance_field(&volume_value_range);
+
+            voxel_cloud1.interpolate_to(&voxel_cloud2, interpolation_factor);
+        } else {
+            let error = FuncError::new(FuncInterpolatedUnionError::EmptyScalarField);
+            log(LogMessage::error(format!("Error: {}", error)));
+            return Err(error);
+        }
```

Comment:
Similar to my previous comment. The union of **2** bounding boxes can not be `None`. This branch will never happen.

Reply:
So, should `BoundingBox::union` even return an `Option`? `None` can't happen at all.

Reply:
It can. If you give it an empty iterator.

Reply:
true